test/NuGet.Core.Tests/NuGet.Frameworks.Test/FrameworkReducerTests.cs

```diff
@@ -110,6 +110,27 @@ public class FrameworkReducerTests
         [InlineData("net6.0-ios11.0", "net6.0-ios12.0,net6.0,net6.0-ios13.0", "net6.0")]
         [InlineData("net7.0-ios", "net6.0,net5.0-ios,netstandard2.1", "net6.0")]
         [InlineData("net7.0-ios", "net6.0,net6.0-ios,netstandard2.1", "net6.0-ios")]
+        // Some net6.0 platforms have "special" fallbacks to xamarin over net5.0
+        [InlineData("net6.0-ios", "xamarin.ios,xamarin.mac,net5.0", "xamarin.ios")]
+        [InlineData("net6.0-maccatalyst", "xamarin.ios,xamarin.mac,net5.0", "xamarin.ios")]
+        [InlineData("net6.0-macos", "xamarin.mac,xamarin.ios,net5.0", "xamarin.mac")]
+        [InlineData("net6.0-tvos", "xamarin.tvos,xamarin.ios,net5.0", "xamarin.tvos")]
+        [InlineData("net6.0-ios", "xamarin.mac,xamarin.ios,net5.0", "xamarin.ios")]
+        [InlineData("net6.0-ios", "xamarin.mac,net5.0", "net5.0")]
+        [InlineData("net6.0-ios", "xamarin.mac", null)]
+        [InlineData("net6.0-maccatalyst", "xamarin.mac", null)]
+        [InlineData("net7.0-ios", "xamarin.mac,net5.0,net6.0", "net6.0")]
+        [InlineData("net7.0-ios", "net6.0,xamarin.ios,xamarin.mac,net5.0", "net6.0")]
+        [InlineData("net7.0-ios", "net7.0-macos,xamarin.mac,xamarin.ios,net6.0,net6.0-ios,net6.0-macos,net5.0", "net6.0-ios")]
+        [InlineData("net7.0-macos", "xamarin.ios,net5.0,net6.0", "net6.0")]
+        [InlineData("net7.0-macos", "net6.0,xamarin.ios,xamarin.tvos,net5.0", "net6.0")]
+        [InlineData("net7.0-macos", "net7.0-ios,xamarin.ios,net6.0,net6.0-ios,net6.0-macos,net5.0", "net6.0-macos")]
+        [InlineData("net7.0-tvos", "xamarin.ios,net5.0,net6.0", "net6.0")]
+        [InlineData("net7.0-tvos", "net6.0,xamarin.ios,xamarin.mac,net5.0", "net6.0")]
+        [InlineData("net7.0-tvos", "net7.0-ios,xamarin.ios,net6.0,net6.0-ios,net6.0-tvos,net5.0", "net6.0-tvos")]
+        [InlineData("net7.0-maccatalyst", "xamarin.mac,net5.0,net6.0", "net6.0")]
+        [InlineData("net7.0-maccatalyst", "net6.0,xamarin.tvos,xamarin.mac,net5.0", "net6.0")]
+        [InlineData("net7.0-maccatalyst", "net7.0-ios,xamarin.mac,net6.0,net6.0-maccatalyst,net6.0-ios,net5.0", "net6.0-maccatalyst")]
```

Comment:
> Individually, each of these tests seems fast enough, but all the unnecessary TFMs being parsed make the tests just that little bit slower, and all tests combined (not just in this PR, but the codebase combined) make running tests locally or on CI slower as well

I'd rather have a product that works correctly, than one that takes 10 mins less on the CI. 

I understand that this is not what you are suggesting, but rather it's about removing redundant data in the tests, but I'd rather have tests that test too much, than not enough. 

We've already noticed people doing some testing gymnastics to compile with the tests should be minimalistic and fast guidance.

Reply:
Ok, so at least one person doesn't agree with the "it would be nice if tests were faster", but what about more understandable tests?  With specifying so many TFMs, it's harder to understand what the test is actually trying to test.

Reply:
I have no problem with cleaning up the unnecessary data.

Reply:
These tests do not have much of a performance impact at all. These are _incredibly_ fast tests. I do not feel it necessary to have fewer test cases.

I also do not want to reduce the complexity of the current cases, because many of them just generally test multiple things which can't be tested in isolation (because they might only happen in more complex cases). The simpler cases are more clear, but the more complex ones are essentially some light fuzzing on my end, and they _have_ found weird bugs while I was developing stuff. I don't think documenting them would be particularly helpful either, tbh.